src/manager.cpp

```diff
@@ -388,8 +388,36 @@ void Manager::collectSingleFruVpd(
 void Manager::deleteSingleFruVpd(
     const sdbusplus::message::object_path& i_dbusObjPath)
 {
-    // Dummy code to supress unused variable warning. To be removed.
-    logging::logMessage(std::string(i_dbusObjPath));
+    std::string l_dbusObjPath{i_dbusObjPath};
+    const nlohmann::json& l_sysCfgJsonObj = m_worker->getSysCfgJsonObj();
+
+    try
+    {
+        if (l_sysCfgJsonObj.empty())
+        {
+            throw std::runtime_error(
+                "System config JSON is not present. deleteFruVpd failed for: " +
+                l_dbusObjPath);
+        }
+
+        const std::string& l_fruPath =
+            jsonUtility::getFruPathFromJson(l_sysCfgJsonObj, l_dbusObjPath);
+
+        if (l_fruPath.empty())
+        {
+            throw std::runtime_error(
+                "DBus object path not present in system config JSON. Cannot perform deletion for: " +
+                l_dbusObjPath);
+        }
+
+        m_worker->deleteFruVpd(i_dbusObjPath);
+    }
+    catch (const std::exception& l_ex)
+    {
+        // ToDo - log PEL
+        logging::logMessage("Failed to delete FRU VPD with error: " +
```

Comment:
can we add for which fru delete failed as well ?

Reply:
Done